bsp/src/mill/bsp/BspCompileProblemReporter.scala

```diff
@@ -130,7 +130,6 @@ class BspCompileProblemReporter(
       pos.endColumn.orElse(pos.pointer).getOrElse[Int](start.getCharacter.intValue())
     )
     val diagnostic = new bsp.Diagnostic(new bsp.Range(start, end), problem.message)
-    diagnostic.setCode(pos.lineContent)
```

Comment:
So we hit on the same thing in Bloop, but I believe this is incorrect. `setCode` is supposed to be used to set the actual unique diagnostic code of the diagnostic, not display the code of the line the diagnostic is one. According to the spec, the BSP `Diagnostic` is the same as the `LSP` [points towards this](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#diagnostic). Even though it's not super clear, the hint is that it can be a `String` or an `Int`.

Reply:
Ideed, quoting your link:

>```
>/**
> * The diagnostic's code, which might appear in the user interface.
> */
>code?: integer | string;
>
>/**
> * An optional property to describe the error code.
> *
> * @since 3.16.0
> */
>codeDescription?: CodeDescription;
>```

Then `code` is more the "404" in "Not Found"-meaning of a error code.